### PR TITLE
WIP fix: do not encode IDN domain

### PR DESCRIPTION
**Pull request by curbengh**
More comprehensive approach to https://github.com/hexojs/hexo-generator-sitemap/pull/61. This PR also encode root value.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -67,4 +68,20 @@ describe('Sitemap generator', () => {
       result.should.be.ok;
     });
   });
+
+  it('IDN handling', () => {
+    hexo.config.url = 'http://fôo.com/bár';
```
Comment (curbengh) [inline test/index.js]: While this option can be applied, but the following `result` still show "http://yoursite.com/".

Comment (curbengh) [reply test/index.js]: I think it's related to `describe()` `it()`. But I don't know how to proceed.

Comment (curbengh) [reply test/index.js]: could also related to how [urlObject](https://nodejs.org/api/url.html#url_legacy_url_api) is referenced.

**Comment by curbengh** [comment]
Pending https://github.com/hexojs/hexo-util/pull/77

**Comment by tomap** [comment]
hexojs/hexo-util#77 merged

**Comment by curbengh** [comment]
Superseded by https://github.com/hexojs/hexo-generator-sitemap/pull/68

**Pull request closed without merging**